packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -186,15 +186,26 @@ class WidgetSpan extends PlaceholderSpan {
   @override
   bool visitDirectChildren(InlineSpanVisitor visitor) => true;
 
-  @override
+    @override
   InlineSpan? getSpanForPositionVisitor(TextPosition position, Accumulator offset) {
-    if (position.offset == offset.value) {
+    // Record where this WidgetSpan begins in the flat text.
+    final int start = offset.value;
+    // Advance past this one “placeholder” code unit.
+    offset.increment(1);
```

Comment:
I think this increment can remain in its original place (above the last null in this method).